src/interface/blas2_interface.hpp

```diff
@@ -55,67 +55,85 @@ typename Executor::policy_t::event_t _gemv_impl(
     index_t _scratchPadSize = 0, index_t _nRowsWG = 0, index_t _nColsWG = 0) {
   typename Executor::policy_t::event_t ret{};
 
-  index_t M = (trn == transpose_type::Normal) ? _M : _N;
-  index_t N = (trn == transpose_type::Normal) ? _N : _M;
+  constexpr int cache_line_size = 64;
+  constexpr int cache_capacity = cache_line_size / sizeof(element_t);
+  constexpr bool is_transposed = trn != transpose_type::Normal;
+
+  index_t local_size = (_localSize == 0)
+                                ? (!is_transposed 
+                                    ? ex.get_policy_handler().get_work_group_size()
+                                    : ex.get_policy_handler().get_work_group_size() / 2)
+                                : _localSize;
 
-  static constexpr auto data_layout_access =
-      Choose<trn == transpose_type::Normal, access_layout,
-             access_layout::col_major, access_layout::row_major>::type;
-  using data_layout_t = typename Layout<data_layout_access>::type;
-  auto mA = make_matrix_view<data_layout_t>(ex, _mA, M, N, _lda);
-  auto vx = make_vector_view(ex, _vx, _incx, N);
-  auto vy = make_vector_view(ex, _vy, _incy, M);
+  const auto x_vector_size = !is_transposed ? _N : _M;
```

Comment:
Again, non-transposed feels default for me so hence the negation and first value of ternary being the default. Negation doesn't cost anything since its a `constexpr`. Readability is not affected in my opinion, but can change if it bothers people that much haha